conda_lock/conda_lock.py

```diff
@@ -716,13 +732,43 @@ def _solve_for_arch(
     return list(conda_deps.values()) + list(pip_deps.values())
 
 
+def convert_structured_metadata_yaml(in_path: pathlib.Path) -> Dict[str, Any]:
+    with in_path.open("r") as infile:
+        metadata = yaml.safe_load(infile)
+    return metadata
+
+
+def update_metadata(to_change: Dict[str, Any], change_source: Dict[str, Any]) -> None:
+    for key in change_source:
+        if key in to_change:
+            logger.warning(
+                f"Custom metadata field {key} provided twice, overwriting value "
+                + f"{to_change[key]} with {change_source[key]}"
+            )
+    to_change.update(change_source)
+
+
+def get_custom_metadata(
+    metadata_yamls: Sequence[pathlib.Path],
+) -> Optional[Dict[str, str]]:
+    custom_metadata_dict: Dict[str, Any] = {}
```

Comment:
```suggestion
) -> Optional[Dict[Any, Any]]:
    custom_metadata_dict: Dict[Any, Any] = {}
```

Reply:
-1 for the same reason.  It basically means that this can no longer be serialized reliably etc

Reply:
Updated for consistency:

```suggestion
) -> Optional[Dict[str, str]]:
    custom_metadata_dict: Dict[str, str] = {}
```
We may want to add runtime validation as well, since the validation error from Pydantic is really confusing.